pkg/api/webassets/webassets.go

```diff
@@ -29,26 +32,64 @@ type EntryPointInfo struct {
 
 var entryPointAssetsCache *dtos.EntryPointAssets = nil
 
-func GetWebAssets(cfg *setting.Cfg) (*dtos.EntryPointAssets, error) {
+func GetWebAssets(cfg *setting.Cfg, license licensing.Licensing) (*dtos.EntryPointAssets, error) {
 	if cfg.Env != setting.Dev && entryPointAssetsCache != nil {
 		return entryPointAssetsCache, nil
 	}
 
-	result, err := readWebAssets(filepath.Join(cfg.StaticRootPath, "build", "assets-manifest.json"))
-	entryPointAssetsCache = result
+	var err error
+	var result *dtos.EntryPointAssets
+
+	cdn := "" // "https://grafana-assets.grafana.net/grafana/10.3.0-64123/"
+	if cdn != "" {
+		result, err = readWebAssetsFromCDN(cdn)
+	}
+
+	if result == nil {
+		result, err = readWebAssetsFromFile(filepath.Join(cfg.StaticRootPath, "build", "assets-manifest.json"))
+		if err == nil {
+			cdn, _ = cfg.GetContentDeliveryURL(license.ContentDeliveryPrefix())
+			if cdn != "" {
+				result.SetContentDeliveryURL(cdn)
+			}
+		}
+	}
 
+	entryPointAssetsCache = result
 	return entryPointAssetsCache, err
 }
 
-func readWebAssets(manifestpath string) (*dtos.EntryPointAssets, error) {
+func readWebAssetsFromFile(manifestpath string) (*dtos.EntryPointAssets, error) {
 	//nolint:gosec
 	bytes, err := os.ReadFile(manifestpath)
 	if err != nil {
 		return nil, fmt.Errorf("failed to load assets-manifest.json %w", err)
 	}
+	return readWebAssets(bytes)
+}
+
+func readWebAssetsFromCDN(baseURL string) (*dtos.EntryPointAssets, error) {
+	response, err := http.Get(baseURL + "public/build/assets-manifest.json")
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = response.Body.Close()
+	}()
+	bytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	dto, err := readWebAssets(bytes)
+	if err == nil {
+		dto.SetContentDeliveryURL(baseURL)
+	}
+	return dto, err
+}
 
+func readWebAssets(bytes []byte) (*dtos.EntryPointAssets, error) {
```

Comment:
I would change this function so it accepts an `io.Reader` instead. This avoids copying the whole body/file in memory.

Then, in `readWebAssetsFromCDN` you can pass `response.Body` (which satisfies `io.Reader`), and in `readWebAssetsFromFile` you have to change `os.ReadFile` with `os.Open` (which returns `*os.File` which satisfies `io.Reader`)


```go
func readWebAssets(r io.Reader) (*dtos.EntryPointAssets, error) {
	// ...
	if err := json.NewDecoder(r).Decode(&manifest); err != nil {
		// ...
	}
```

Reply:
nice -- good idea!